common/src/main/java/com/microsoft/identity/common/internal/broker/BoundServiceClient.java

```diff
@@ -163,8 +163,14 @@ public BoundServiceClient(@NonNull final Context context,
      * Disconnects (unbinds) from the service.
      */
     public void disconnect() {
+        final String methodTag = TAG + ":disconnect";
         if (mHasStartedBinding) {
-            mContext.unbindService(mConnection);
+            try {
+                mContext.unbindService(mConnection);
+            } catch (final IllegalArgumentException e) {
```

Comment:
May I ask what was the cause of this error? why IllegalArgumentException specifically?

Reply:
We only have the callstacks rather than the exception details, but it's one of the three coming from here, relating to trying to bind things that are unbound already:
https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/app/LoadedApk.java#1945

As this is a cleanup function, there isn't much of an action to take here.  It wasn't clear to me how we might be getting in one of these three states -- perhaps registration is failing early enough that unbinding isn't actually necessary?

Reply:
Could you please add the context (as much as possible) as a comment in this error block?  (i.e. OneAuth's error stats, etc) - so that future devs are aware what this error is from.

It could be that the connection was terminated mid-way (by android itself) and the attempt to unbind therefore failed.